src/services/users/create-new-user.test.js

```diff
@@ -0,0 +1,78 @@
+const { expect } = require('chai')
+const { faker } = require('@faker-js/faker')
+const db = require('../../db')
+const {fixtures} = require('../../../test/utils')
+const createNewUser = require('./create-new-user')
+
+
+describe('Create a New User', () => {
+	let signupInfo
+	let userInfo = {}
+	
+
+	beforeEach(async () => {
+		userInfo = {
+			firstName: faker.name.firstName(),
+			lastName: faker.name.lastName(),
+			email: faker.internet.email(),
+			password: faker.internet.password()
+		}
+		signupInfo = await fixtures.insertUserInvite()
+	})
+
+	it('should throw an error if the invite is invalid', async () =>
+
+		expect(createNewUser({...userInfo}))
+			.to.be.rejectedWith(
+				'Email is not present in invites list')
+	)
+    
+	it('should throw an error if user already exists', async () =>{
+		const { email } = signupInfo
+		await fixtures.insertUserInvite({email})
+		await fixtures.updateInviteStatus({email})
+		return expect(createNewUser({
+			firstName:userInfo.firstName,
+			lastName:userInfo.lastName, email, 
+			password:userInfo.password}))
+			.to.be.rejectedWith(
+				'User already exists')
+	})
+
+
+	it('should create a new user', async () =>{
+		const { email } = signupInfo
+		await createNewUser({
+			firstName:userInfo.firstName, 
+			lastName:userInfo.lastName, email, 
+			password:userInfo.password})
+
+		const {rows} = await db.query(
+			`SELECT * FROM users
+             WHERE email = $1`,[email] )
+
+		const createdUser = rows[0]
+		
+		return expect(createdUser).to.exist	&& 
+			expect(createdUser.firstName).to.equal(userInfo.firstName)
+
+	})
+
+	it('should return the right data', async () => {
+		const { email } = signupInfo
+		
+		const newUser = await createNewUser({
+			firstName:userInfo.firstName, 
+			lastName:userInfo.lastName, email, 
+			password:userInfo.password})
+
+		expect(newUser).to.have
```

Comment:
yea getting closer, but we want to know that the returned data is the same, not just the properties, else  the  `refreshToken` for instance can be returned as undefined and test will still pass

so we will still make a query for the inserted user like you're doing in the other test case and `expect(newUser)` to equal the data from the createdUser (the one from the query)

```js
expect(newUser).to.eql({
			accessToken, //generate from the queried user data
			refreshToken: createdUser.refreshToken,
			userId: createdUser.id
		})
```